### PR TITLE
Proxy Route Tasks

### PR DESCRIPTION
**Pull request by lancetipton**
## Context

* Sometimes it's a pain to figure out the url to a running application
* Currently we use the Traefik Admin UI, but there are no direct links to the apps
  * We have to go to the Admin UI in the borwser
  * Find the container route, and navigate to its page
  * Then copy and paste the url from the displayed rule, which is not always super clear

## Goal
* Make it easy to open apps in a browser that are registered to the proxy

## Updates
* Added to tasks to the proxy tasks that allow for opening a registered proxy route
  * `keg proxy list` => Lists all registered routes
    * Hides all internal routes by default, except for the the Admin UI route
    * Allows a filter option to filter the results
      * Using the filter includes internal routes
  * `keg proxy open` => Opens a route's url in the browser
    * Allows a context option to define the route to be open
    * If no context is passed, the user is asked to select a route from a list
      * Hides all internal routes by default, except for the the Admin UI route
      * If the context matches an internal route, it will be displayed

## Testing
* **TODO REMINDER**
  * This is to remind me to write tests for the proxy util methods added in the PR
  * If reviewing this PR, you can skip down the the **Setup** section below

**Setup**
  * Pull this pr => `keg cli && keg pr 62`
  * Start multiple taps/apps to allow testing
    * `keg evf start`
    * `keg retheme start`
    * `keg components start`

**Test 1**
* After starting a few taps / apps, run the cmd => `keg proxy ls` || `keg proxy list`
  * This should show a list of the registered routes
  * [Looks like this](https://prnt.sc/11xgmin)
  * Ensure the correct routes are displayed
* Next run the command => `keg proxy list components`
  * Ensure only show keg-components routes are displayed
  * [Looks like this](https://prnt.sc/11xgnec)
* Next run the command => `keg proxy list api`
  * Ensure only shows the internal api route
    * This is useless in our case, so you can see why I hid it by default
  * [Looks like this](https://prnt.sc/11xgovp)
* Try other filter combinations as needed

**Test 2**
* Run the cmd => `keg proxy open` || `keg proxy op`
  * Ensure is displays a list of available routes to open
  * [Looks like this](https://prnt.sc/11xgpfc)
* Select one of the routes
  * Enter the index of the route from the displayed list and press enter
* Ensure it then opens the selected route in your default browser
  * It uses the npm package [open](https://www.npmjs.com/package/open) to open the browser
  * Which means Firefox should work if it's your default, but I have not tested it
* Run the cmd again, but this time pass the linked name of one of the running taps or application
  * For example if `keg-components` is running, run the cmd => `keg proxy open components`
    * This should open `keg-components` in the browser
  * If the `evf-tap` is running, run the cmd => `keg proxy open evf`
    * This should open the `evf-tap` in the browser

### CODE REVIEW

File: src/tasks/proxy/list.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    groups[name] = groups[name] || []
+    groups[name].push({
+      branch: `    * ${rest.join('-')}`,
+      url: `    * https://${item.rule.split('`')[1]}`
+    })
+  })
+
+  // Print each group to the terminal
+  mapObj(groups, (name, items) => {
+    Logger.yellow(`  ${ wordCaps(name) }`)
```
Comment (mpcarolin) [inline src/tasks/proxy/list.js:27]: Super minor and not at all blocking, but any chance we can make the titles of containers blue or green? lol

I associate yellow with warnings.

**Review by mpcarolin — APPROVED** [review]
Looks good and super helpful. 

I left one very subjective and non-blocking request. Just ignore it if you prefer it the way it is; I won't really care.

File: src/tasks/proxy/open.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+const askForRoute = async items => {
+  const index = await ask.promptList(
+    items.map(item => {
+      const [ name, ...rest ] = item.service.split('-')
+      const branch = rest.length ? `-${rest.join('-')}\n` : '\n'
+      const url = `       * http://${item.rule.split('`')[1]}\n`
+
+      return `${wordCaps(name)}${branch}${url}`
+    }),
+    'Proxy Routs: ( Route | URL )\n',
```
Comment (mpcarolin) [inline src/tasks/proxy/open.js:23]: typo on `Routs`

**Review by mpcarolin — APPROVED** [review]
Also saw a typo after I approved. Still approved, but just pointing this one out for you

File: src/tasks/proxy/list.js (hunk 1 above)
Comment (lancetipton) [reply src/tasks/proxy/list.js:27]: Update to green

File: src/tasks/proxy/open.js (hunk 2 above)
Comment (lancetipton) [reply src/tasks/proxy/open.js:23]: updatd

**Pull request merged**